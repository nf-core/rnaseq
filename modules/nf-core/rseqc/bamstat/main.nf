process RSEQC_BAMSTAT {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rseqc:5.0.3--py39hf95cd2a_0' :
        'biocontainers/rseqc:5.0.3--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.bam_stat.txt"), emit: txt
    path  "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    bam_stat.py \\
        -i $bam \\
        $args \\
        > ${prefix}.bam_stat.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rseqc: \$(bam_stat.py --version | sed -e "s/bam_stat.py //g")
    END_VERSIONS
    """
}
