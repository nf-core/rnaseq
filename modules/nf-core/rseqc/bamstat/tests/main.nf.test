nextflow_process {

    name "Test Process RSEQC_BAMSTAT"
    script "../main.nf"
    process "RSEQC_BAMSTAT"

    tag "modules"
    tag "modules_nfcore"
    tag "rseqc"
    tag "rseqc/bamstat"

    config "./nextflow.config"

    test("sarscov2 - [meta] - bam") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = [ 
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam')
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
