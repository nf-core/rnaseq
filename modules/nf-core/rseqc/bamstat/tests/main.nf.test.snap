{
    "sarscov2 - [meta] - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam_stat.txt:md5,2675857864c1d1139b2a19d25dc36b09"
                    ]
                ],
                "1": [
                    "versions.yml:md5,58f74a7ff9d2966142c81a4a4735dbf3"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam_stat.txt:md5,2675857864c1d1139b2a19d25dc36b09"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,58f74a7ff9d2966142c81a4a4735dbf3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2023-12-14T17:00:23.830276754"
    }
}